### PR TITLE
feat: add support for span id and custom key for trace id and span id for otelzap

### LINKED ISSUE #109: Add span_id with trace_id for zap
According to OTel spec, we can add `span_id` to the log to link a log to a specific span not just a trace https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-spanid

### PR DESCRIPTION
**Pull request by pragmaticivan**
fixes #109

### CODE REVIEW

File: otelzap/otelzap.go [hunk 1]
```diff
@@ -439,7 +450,7 @@ func (s *SugaredLogger) FatalfContext(ctx context.Context, template string, args
 func (s *SugaredLogger) logArgs(
 	ctx context.Context, lvl zapcore.Level, template string, args []interface{},
 ) {
-	if lvl < s.l.minLevel {
+	if lvl <= s.l.minLevel {
```
Comment (pragmaticivan) [inline otelzap/otelzap.go:453]: The current code for this line is causing some weird behaviors when the log level is `debug`

**Comment by vmihailenco** [comment]
Thanks for the PR, but otelzap now uses Otel Logs API which has top-level fields for trace & span ids.

**Comment by pragmaticivan** [comment]
Hi @vmihailenco could you clarify that? this PR is backward compatible, it happens that many vendors don't follow the same API `yet` and this would be required to support New Relic for example.

**Comment by vmihailenco** [comment]
@pragmaticivan otelzap now does not inject trace_id/span_id and instead relies on Otel Logs API to do it automatically: https://github.com/uptrace/opentelemetry-go-extra/blob/main/otelzap/otelzap.go#L164-L196

And Otel Logs API will set trace/span ids on the `log.Record` structure directly without using attributes.

**Comment by vmihailenco** [comment]
>would be required to support New Relic for example.

New Relic would just need to support OpenTelemetry Logs API.

**Pull request closed without merging**